### PR TITLE
Set local_time column on time table to hidden

### PR DESCRIPTION
**Pull request by theopolis**
The concept of a "local" UNIX time is confusing since UNIX time should be counted from the epoch, which specifies a timezone.

Fixes: #6743
Fixes: #7274

### CODE REVIEW

**Review by sharvilshah — APPROVED** [review]

**Comment by directionless** [comment]
i won't block this -- but if you can please hold up? I want to try to understand what the intent here had been

**Comment by theopolis** [comment]
Check out @Smjert's analysis in the linked issue here: https://github.com/osquery/osquery/issues/7274

**Review by directionless — CHANGES_REQUESTED** [review]
I spent a bit staring at this, and trying to understand the past. Notable here is #2504 which claimed to remove local time. And @Smjert's writeup 

Given that this is landing in 5.0, I think we should be more extensive -- if we think the column does nothing, we should simply drop it and be done. It doesn't seem worth carrying the baggage forward. I think we should also drop the `timezone` column. Or merge it with local_timezone. Or whatever. 

My read on `--utc` is that it adjust the human readable parts of the time table to utc vs local time. I think there's value there, but that the help on the flag should be updated. The whole human readable thing feels a little duplicated with the sqlite time functions, but :shrug: it seems fine.

**Comment by theopolis** [comment]
I agree with that analysis and I'll make some changes to be more "breaking". I think removing the `--utf` flag is best and going with the SQLite time functions is the most universal.

**Comment by theopolis** [comment]
I opted to keep `"timezone"` to indicate that the human-readable columns here referred to time in UTC.

File: specs/utility/time.table [hunk 1]
```diff
@@ -9,11 +9,8 @@ schema([
     Column("minutes", INTEGER, "Current minutes in the system"),
     Column("seconds", INTEGER, "Current seconds in the system"),
     Column("timezone", TEXT, "Current timezone in the system"),
```
Comment (directionless) [inline specs/utility/time.table:11]: This is now hardcoded, not the system timezone. I'd probably change the desc of all of these, they're not system time. They're in UTC

File: specs/utility/time.table [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
-    Column("local_time", INTEGER, "Current local UNIX time in the system",
-        aliases=["localtime"]),
-    Column("local_timezone", TEXT, "Current local timezone in the system"),
-    Column("unix_time", INTEGER,
-        "Current UNIX time in the system, converted to UTC if --utc enabled"),
-    Column("timestamp", TEXT, "Current timestamp (log format) in the system"),
-    Column("datetime", TEXT, "Current date and time (ISO format) in the system",
+    Column("weekday", TEXT, "Current weekday in UTC"),
+    Column("year", INTEGER, "Current year in UTC"),
+    Column("month", INTEGER, "Current month in UTC),
```
Comment (sharvilshah) [inline specs/utility/time.table:6]: @theopolis This is missing an ending `"` in `"Current month in UTC` (that's also the reason the build is failing)

**Review by sharvilshah — DISMISSED** [review]

**Comment by mike-myers-tob** [comment]
Moving this to 5.1 because we want to get 5.0.1 out. As discussed in Slack recently. :-)

**Comment by theopolis** [comment]
@directionless and @sharvilshah, ready for review and merge?

File: specs/utility/time.table [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
-    Column("timestamp", TEXT, "Current timestamp (log format) in the system"),
-    Column("datetime", TEXT, "Current date and time (ISO format) in the system",
+    Column("weekday", TEXT, "Current weekday in UTC"),
+    Column("year", INTEGER, "Current year in UTC"),
+    Column("month", INTEGER, "Current month in UTC"),
+    Column("day", INTEGER, "Current day in UTC"),
+    Column("hour", INTEGER, "Current hour in UTC"),
+    Column("minutes", INTEGER, "Current minutes in UTC"),
+    Column("seconds", INTEGER, "Current seconds in UTC"),
+    Column("timezone", TEXT, "Current timezone in UTC"),
```
Comment (directionless) [inline specs/utility/time.table:11]:
```suggestion
    Column("timezone", TEXT, "Timezone for reported time(UTC)"),
```

**Review by directionless — COMMENTED** [review]
Small nit, though I'd take it either way

**Comment by directionless** [comment]
(Trigger a new ci build)

**Review by directionless — APPROVED** [review]

**Pull request merged**